lib/SimpleSAML/Auth/Source.php

```diff
@@ -218,6 +218,116 @@ public function initLogin($return, $errorURL = null, array $params = [])
     }
 
 
+    /**
+     * Register a logout callback to be called after the consumer of this authentication source is logged out.
+     *
+     * @param $callback Callback structure ([CLASS_NAME, STATIC_FUNCTION_NAME]).
+     *        Static function has to have following signature: ($param1, ..., $paramN, string $returnUrl) with
+     *        $param1-N being members of $params.
+     *        If callback does not return directly (i.e. redirect) it has to make sure that $returnUrl is called after
+     *        processing finished.
+     * @param array $params Arbitrary parameters.
+     * @throws \Exception
+     */
+    public function registerAsConsumerLogoutCallback($callback, $params = []) {
+        assert(is_callable($callback));
+        assert(is_array($params));
```

Comment:
We can use type hints here and a return type instead of the assertions. This also goes for other methods: use type hints/return types, and remove assertions that only assert the type.

Reply:
Fixed in 7693d74c2d15801603ea39ab6b0a0aaf78122f45